### ISSUE TITLE
[batch] Duration of batch experiment increases greatly for the first few rounds when using multiple threads

### ISSUE THREAD

**Issue by lesquoyb** [gama-platform/gama#290]
**Describe the bug**
I run the exact same experiment 100 times using batch experiments and a repeat value of 3 and here is the total duration of each experiment during that batch:
![image](https://github.com/user-attachments/assets/62c9ecc7-6500-434b-a430-c52f0611ecb3)
we can see that the two first rounds of 3 simulations were faster than the rest, and the second round way already slower than the first, this is a consistent result I have after running this kind of experiment many times.
What is weird is that the exact same batch but with a value of 1 for the repeat facet yields execution times that varies only a little since the beginning:
![image](https://github.com/user-attachments/assets/cda4820f-68ac-4dde-85a8-fa1082a9122d)
Here it's the 10 first execution and the duration has variations of about 10/20% contrary to the more than 100% we can observe in the previous case

So I think there's something fishy here and there's probably a way we can improve greatly execution time in batch.

**To Reproduce**
1. Run an experiment in batch with a repeat value greater than 1
2. make the simulation save their duration at the end of it
3. Observe that the duration raises for the first few rounds then (probably depending on your simulation) it stabilizes
4. change the repeat to 1 and observe that it doesn't increase anymore

**Expected behavior**
Somewhat constant simulation duration

**Comment by AlexisDrogoul** [comment]
Would you have a test model (not MAELIA :)) to reproduce this behavior ?

**Comment by lesquoyb** [comment]
with the fix of #291 it seems less visible:
![image](https://github.com/user-attachments/assets/3f29e55c-7672-4103-8760-e58fda9ac3b5)

The code is any batch with parallel > 1, so for example:
```

model batchtest

global {
	float start <- gama.machine_time;
	float seed <- 3.1415;
	init {
		gama.pref_csv_separator <- ";";
	}

	
	reflex d {
		int size <- 250;
		let m <- matrix_with({size,size}, rnd(0,255));
		m <- shuffle(m);
		loop i from:0 to:size-1 {
			loop j from:0 to:size-1 {
				m[i,j] <- rnd(255);
			}
		}
	}
	
	reflex s when:cycle=49{
		save [int(self), int(gama.machine_time-start)] to:"data.csv" rewrite:false;
	}

}

experiment b type:batch until:cycle>=100 repeat:50 parallel:3 keep_simulations:false{
		

	
}

```

**Comment by AlexisDrogoul** [comment]
I've run 1000 simulations of the test model and the pattern is not here anymore. Instead we have a quite remarkable stability (except one episode, where I manipulated the UI a little bit). 

<img width="1645" alt="image" src="https://github.com/user-attachments/assets/0e9debcd-f2e6-4ba4-b545-7c22866cb87c">

So I would be in favor of closing this issue.
